### DISCUSSION TITLE
-bash: /home/linuxbrew/.linuxbrew/bin/fd: cannot execute binary file: Exec format error

### DISCUSSION THREAD

**Discussion by Trid-collab** [Homebrew/discussions#3344] [Linux]
I am doing a fresh install of brew on my WSL2 setup. After installing fd, exa, ripgrep and trying to run the these commands I am getting the following errors:

-bash: /home/linuxbrew/.linuxbrew/bin/fd: cannot execute binary file: Exec format error
-bash: /home/linuxbrew/.linuxbrew/bin/exa: cannot execute binary file: Exec format error
-bash: /home/linuxbrew/.linuxbrew/bin/ripgrep: cannot execute binary file: Exec format error

Whereas the other installs like git or bpytop or mc has been working fine without any issues

Is there something I am missing here

**Comment by jonchang** [comment]
Are you certain you're using WSLv2? We've only seen this issue with Rust-compiled binaries on WSLv1. If you look at [this issue](https://github.com/Homebrew/linuxbrew-core/issues/20393) (and linked [issues](https://github.com/Homebrew/linuxbrew-core/issues/21293)) you'll find that upgrading to WSLv2 seemed to fix the problem. It's not clear why this occurs but there are some ideas in that issue thread. You could try to `brew reinstall --build-from-source ripgrep` but this will take a while and isn't really supported, so if you encounter issues the Homebrew maintainers won't be able to help.

**Comment by Trid-collab** [comment]
Yes, I am on WSL2. I have also tried brew reinstall --build-from-source fd without any success. It still had the same error. 
The other thing I noticed that I have another laptop with WSL2 in which I had installed fd, ripgrep,exa with brew with out any issues. The only difference I guess being that those are the older versions which I had installed sometime back say about a month back
